src/Sylius/Component/Core/Model/Channel.php

```diff
@@ -74,6 +74,8 @@ class Channel extends BaseChannel implements ChannelInterface
     /** @var bool */
     protected $accountVerificationRequired = true;
 
+    protected bool $shippingAddressInCheckoutRequired = false;
```

Comment:
Maybe it would be more natural to make it a `string` argument (enum would be great :D but not yet :dancer:)? `string $defaultCheckoutAddress = AddressType::BILLING`? Would be also probably more extendable (if more than 2 addresses are used in the store)

Reply:
I thought about it, but I decided that it would be too exaggerated. The usage of the flag meets the current business requirements, its implementation is still much simpler and I don't think we need this kind of potential customization 🤔

Reply:
**We** do not need it, but someone using Sylius could need 😄 Moreover, the name of the flag is quite misleading, because, as far as I see, it's not about the requirement of shipping address in checkout or not, but about which of the addresses is used as a _base_ address. Also, making it an enum-like string would not add too much complexity (it's just a change of the property type with some potential simple validation in the `setBaseAddressType(string $type)` method, but would make Behat implementation event simpler (we could change the toggle to the select box with 2 predefined types 🚀). I generally feel that the business requirement is about the **type** of the default checkout address, but we use a **flag** to reflect it in the code.

Still, we can continue this discussion in another PR, we have some time to potentially iterate over it before 1.12.0 release 🖖

Reply:
Yup, I would be for leaving this PR with the current approach and potentially continue the refactor in another PR 👌🏻